### ISSUE TITLE
ResourceExhaustedError (see above for traceback): OOM when allocating tensor of shape [] and type float

### ISSUE THREAD

**Issue by xiaoxiaoAurora** [allenai/bilm-tf#195]
ResourceExhaustedError (see above for traceback): OOM when allocating tensor of shape [] and type float
         [[node lm/RNN_0/rnn/multi_rnn_cell/cell_0/lstm_cell/kernel/Initializer/random_uniform/min (defined at /home/miniconda3/lib/python3.6/site-packages/bilm/training.py:410)  = Const[_class=["loc:@lm/RNN_0/rnn/multi_rnn_cell/cell_0/lstm_cell/kernel/Assign"], dtype=DT_FLOAT, value=Tensor<type: float shape: [] values: -0.0185652673>, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]

How can I solved this? 

1. vocabulary size: 146, 828 (including `<S>, </S>, <UNK>`)
2. I have only one GPU and changed the followings:  

- set ` n_gpu=1`, 
- export CUDA_VISIBLE_DEVICES=0
3.  NVIDIA-SMI 418.43       Driver Version: 418.43       CUDA Version: 10.1

**Comment by matt-peters** [comment]
Decrease the batch size or other hyperparameters (vocabulary size, model size, number of negative samples for the softmax) until it fits onto your GPU.
